packages/enzyme/src/selectors.js

```diff
@@ -0,0 +1,315 @@
+import { createParser } from 'rst-selector-parser';
+import isEmpty from 'lodash/isEmpty';
+import unique from 'lodash/uniq';
+import {
+  treeFilter,
+  nodeHasId,
+  findParentNode,
+  nodeMatchesObjectProps,
+  childrenOfNode,
+  hasClassName,
+} from './RSTTraversal';
+import { nodeHasType, nodeHasProperty } from './Utils';
+// our CSS selector parser instance
+const parser = createParser();
+
+// Combinators that allow you to chance selectors
+const CHILD = 'childCombinator';
+const ADJACENT_SIBLING = 'adjacentSiblingCombinator';
+const GENERAL_SIBLING = 'generalSiblingCombinator';
+const DESCENDANT = 'descendantCombinator';
+
+// Selectors for targeting elements
+const SELECTOR = 'selector';
+const TYPE_SELECTOR = 'typeSelector';
+const CLASS_SELECTOR = 'classSelector';
+const ID_SELECTOR = 'idSelector';
+const ATTRIBUTE_PRESENCE = 'attributePresenceSelector';
+const ATTRIBUTE_VALUE = 'attributeValueSelector';
+// @TODO we dont support these, throw if they are used
+const PSEUDO_CLASS = 'pseudoClassSelector';
+const PSEUDO_ELEMENT = 'pseudoElementSelector';
+
+/**
+ * Calls reduce on a array of nodes with the passed 
+ * function, returning only unique results.
+ * @param {Function} fn 
+ * @param {Array<Node>} nodes 
+ */
+function uniqueReduce(fn, nodes) {
+  return unique(nodes.reduce(fn, []));
+}
+
+/**
+ * Takes a CSS selector and returns a set of tokens parsed
+ * by scalpel.
+ * @param {String} selector 
+ */
+function safelyGenerateTokens(selector) {
```

Comment:
feels weird to call this `safely{thing}` but still have it throw

Reply:
Nah throwing is the safest thing ever! Unsafe things are what don't throw :-)

Reply:
Yeah, I was thinking "safe" because we explicitly throw our own error instead of letting the parser crash, and know that the tokens don't require additional validation.